Mage.Sets/src/mage/cards/s/SunfrillImitator.java

```diff
@@ -0,0 +1,120 @@
+package mage.cards.s;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetCreaturePermanent;
+import mage.util.functions.CopyApplier;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class SunfrillImitator extends CardImpl {
+
+    public SunfrillImitator(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{G}");
+        
+        this.subtype.add(SubType.DINOSAUR);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Whenever Sunfrill Imitator attacks, you may have it become a copy of another target Dinosaur you control, 
+        // except its name is Sunfrill Imitator and it has this ability.
+        this.addAbility(new SunfrillImitatorAbility());
+    }
+
+    private SunfrillImitator(final SunfrillImitator card) {
+        super(card);
+    }
+
+    @Override
+    public SunfrillImitator copy() {
+        return new SunfrillImitator(this);
+    }
+}
+
+class SunfrillImitatorAbility extends AttacksTriggeredAbility {
+
+    public SunfrillImitatorAbility() {
+        super(
+            new SunfrillImitatorEffect(),
+            true,
+            "Whenever Sunfrill Imitator attacks, you may have it become a copy of another target Dinosaur " +
+                    "you control, except its name is Sunfrill Imitator and it has this ability."
+        );
+        FilterCreaturePermanent dino_filter = new FilterCreaturePermanent(SubType.DINOSAUR, "dinosaur");
+        dino_filter.add(AnotherPredicate.instance);
+        this.addTarget(new TargetCreaturePermanent(dino_filter));
+    }
+
+    private SunfrillImitatorAbility(final SunfrillImitatorAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
```

Comment:
I don't think you should be overriding methods to duplicate code defined in the super class

(also ugh, I don't know why we have `ATTACKER_DECLARED` separate from `DECLARED_ATTACKERS`...)

Reply:
My bad, i didnt catch that when i copied this in. What is the difference in between those two enums in terms of behavior? I can't seem to find a concrete answer just looking through the usages.

Reply:
My guess is that `DECLARED_ATTACKERS` is fired once per declare attackers step, and `ATTACKER_DECLARED` is fired once for each attacking creature. In this case either can be used, but for other purposes you might need one or the other.